cmd/command/swap_in.go

```diff
@@ -316,14 +316,35 @@ func QueryRecoverProcessAction(ctx *cli.Context) error {
 		println(err.Error())
 		return err
 	}
-	spClient := utils.MakeGfSpClient(cfg)
+
 	gvgID := ctx.Uint64(gvgIDFlag.Name)
 	gvgfID := ctx.Uint64(vgfIDFlag.Name)
+
+	if gvgfID != 0 {
+		// get client
+		chainClient, err := utils.MakeGnfd(cfg)
+		if err != nil {
+			println(err.Error())
+			return err
+		}
+		familyInfo, err := chainClient.QueryVirtualGroupFamily(ctx.Context, uint32(gvgfID))
+		if err != nil {
+			println(err.Error())
+			return err
+		}
+		if len(familyInfo.GetGlobalVirtualGroupIds()) == 0 {
+			println("recover gvg family complete")
```

Comment:
When query the progress, need to confirm the primary SP of empty family has been replaced by sp itself.